File: apps/watch-wear/app/src/main/java/com/holypadel/wear/ui/Screens.kt
package com.holypadel.wear.ui

import androidx.compose.foundation.Canvas
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.pager.HorizontalPager
import androidx.compose.foundation.pager.rememberPagerState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.alpha
import androidx.compose.ui.draw.clip
import androidx.compose.ui.geometry.CornerRadius
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.geometry.Size
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.Path
import androidx.compose.ui.graphics.StrokeCap
import androidx.compose.ui.graphics.drawscope.DrawScope
import androidx.compose.ui.graphics.drawscope.Stroke
import androidx.compose.ui.unit.dp
import kotlin.math.cos
import kotlin.math.sin
import com.holypadel.wear.MatchState
import com.holypadel.wear.Phase

/** Routes to the right screen purely from the mirrored [MatchState.phase]. */
@Composable
fun WatchApp(
    state: MatchState,
    liveBpm: Int,
    onScore: (String) -> Unit,
    onUndo: () -> Unit,
    onStartLast: () -> Unit,
    onPause: () -> Unit,
    onStop: () -> Unit,
    onCancel: () -> Unit,
    onEnd: () -> Unit,
) {
    Box(
        modifier = Modifier.fillMaxSize().background(CourtColors.Black),
        contentAlignment = Alignment.Center,
    ) {
        when (state.phase) {
            Phase.IDLE -> IdleScreen(state, onStartLast)
            Phase.LIVE -> LiveScoreScreen(state, liveBpm, onScore, onUndo, onPause, onStop, onCancel)
            Phase.WON -> MatchWonScreen(state, onEnd)
        }
    }
}

@Composable
private fun IdleScreen(state: MatchState, onStartLast: () -> Unit) {
    Column(
        modifier = Modifier.fillMaxSize().padding(horizontal = 22.dp, vertical = 20.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.Center,
    ) {
        LabelText("NO LIVE MATCH", sizeSp = 11)
        state.last?.let { last ->
            Spacer(Modifier.height(10.dp))
            Row(verticalAlignment = Alignment.CenterVertically) {
                ResultBadge(last.won, sizeDp = 20)
                Spacer(Modifier.width(8.dp))
                BodyText("Last · ${last.line}", color = CourtColors.White45, sizeSp = 12)
            }
        }
        Spacer(Modifier.height(18.dp))
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .clip(RoundedCornerShape(28.dp))
                .background(CourtColors.Lime)
                .clickable { onStartLast() }
                .padding(vertical = 12.dp),
            horizontalAlignment = Alignment.CenterHorizontally,
        ) {
            DisplayText("START MATCH", sizeSp = 16, color = CourtColors.Ink)
            state.last?.let {
                LabelText("LAST LINEUP", color = CourtColors.Ink, sizeSp = 9)
            }
        }
        Spacer(Modifier.height(10.dp))
        BodyText("or set up on phone", color = CourtColors.White40, sizeSp = 10)
    }
}

/**
 * Two sideways pages, like the Wear/Apple workout app: swipe between the
 * **controls** (undo, pause, stop-and-save, cancel) and the **score face** (tap a
 * team to score). Separating them keeps the score clean and makes an accidental
 * stop impossible while you're tapping points every rally.
 */
@Composable
private fun LiveScoreScreen(
    state: MatchState,
    liveBpm: Int,
    onScore: (String) -> Unit,
    onUndo: () -> Unit,
    onPause: () -> Unit,
    onStop: () -> Unit,
    onCancel: () -> Unit,
) {
    // Land on the score face (page 1) — the surface you touch every rally.
    val pagerState = rememberPagerState(initialPage = 1) { 2 }
    Box(modifier = Modifier.fillMaxSize()) {
        HorizontalPager(state = pagerState, modifier = Modifier.fillMaxSize()) { page ->
            if (page == 0) {
                ControlsPage(state, onUndo, onPause, onStop, onCancel)
            } else {
                ScorePage(state, liveBpm, onScore)
            }
        }
        PagerDots(
            count = 2,
            current = pagerState.currentPage,
            modifier = Modifier.align(Alignment.BottomCenter).padding(bottom = 3.dp),
        )
    }
}

/** The score face — nothing but the two tappable team halves plus a compact header. */
@Composable
private fun ScorePage(state: MatchState, liveBpm: Int, onScore: (String) -> Unit) {
    val paused = state.paused
    Column(modifier = Modifier.fillMaxSize().padding(horizontal = 20.dp, vertical = 14.dp)) {
        // Header: clock (or live bpm) · set/games · LIVE (or PAUSED)
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically,
        ) {
            if (liveBpm > 0) {
                BodyText("$liveBpm♥", color = CourtColors.Lime, sizeSp = 12)
            } else {
                BodyText(state.clock, color = CourtColors.White45, sizeSp = 12)
            }
            LabelText(listOf(state.setLabel, state.games).filter { it.isNotEmpty() }.joinToString("  "))
            Row(verticalAlignment = Alignment.CenterVertically) {
                Dot(sizeDp = 6, color = if (paused) CourtColors.White45 else CourtColors.Lime)
                Spacer(Modifier.width(4.dp))
                LabelText(
                    if (paused) "PAUSED" else "LIVE",
                    color = if (paused) CourtColors.White45 else CourtColors.Lime,
                    sizeSp = 10,
                )
            }
        }

        // While paused, dim the halves and swallow taps — the phone rejects points
        // when paused anyway, but this keeps the watch honest (docs/watch-sync.md).
        val scoreEnabled = !paused
        TeamRow(
            modifier = Modifier
                .weight(1f)
                .alpha(if (paused) 0.35f else 1f)
                .clickable(enabled = scoreEnabled) { onScore("A") },
            short = state.teamA.short,
            point = state.pointA,
            serving = state.teamA.serving,
        )
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .height(1.dp)
                .background(CourtColors.White25),
        )
        TeamRow(
            modifier = Modifier
                .weight(1f)
                .alpha(if (paused) 0.35f else 1f)
                .clickable(enabled = scoreEnabled) { onScore("B") },
            short = state.teamB.short,
            point = state.pointB,
            serving = state.teamB.serving,
        )
    }
}

/** The actions face — undo, pause/resume, the square stop-and-save, and a guarded full cancel. */
@Composable
private fun ControlsPage(
    state: MatchState,
    onUndo: () -> Unit,
    onPause: () -> Unit,
    onStop: () -> Unit,
    onCancel: () -> Unit,
) {
    val paused = state.paused
    var confirmingCancel by remember { mutableStateOf(false) }
    Column(
        modifier = Modifier.fillMaxSize().padding(horizontal = 18.dp, vertical = 14.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(9.dp, Alignment.CenterVertically),
    ) {
        LabelText(
            if (paused) "PAUSED" else "CONTROLS",
            color = if (paused) CourtColors.White45 else CourtColors.Lime,
            sizeSp = 11,
        )
        Row(
            horizontalArrangement = Arrangement.spacedBy(18.dp, Alignment.CenterHorizontally),
            verticalAlignment = Alignment.CenterVertically,
        ) {
            LabeledControl("UNDO") {
                CircleIconButton(ControlIcon.UNDO, onClick = onUndo, enabled = !paused)
            }
            LabeledControl(if (paused) "RESUME" else "PAUSE") {
                CircleIconButton(
                    if (paused) ControlIcon.PLAY else ControlIcon.PAUSE,
                    onClick = onPause,
                    accent = true,
                    diameterDp = 44,
                )
            }
        }
        if (confirmingCancel) {
            // Inline confirm — no dialog dependency; discard is one deliberate tap away.
            LabelText("DISCARD? SCORE LOST", color = CourtColors.White45, sizeSp = 9)
            Row(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
                PillButton("DISCARD", CourtColors.White.copy(alpha = 0.14f), CourtColors.White, onCancel)
                PillButton("KEEP", CourtColors.Lime, CourtColors.Ink) { confirmingCancel = false }
            }
        } else {
            // Stop AND save — the filled square, "court time's up, don't lose the score".
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .clip(RoundedCornerShape(12.dp))
                    .background(CourtColors.Lime)
                    .clickable { onStop() }
                    .padding(vertical = 9.dp),
                horizontalArrangement = Arrangement.spacedBy(6.dp, Alignment.CenterHorizontally),
                verticalAlignment = Alignment.CenterVertically,
            ) {
                Box(Modifier.size(12.dp).clip(RoundedCornerShape(2.dp)).background(CourtColors.Ink))
                DisplayText("STOP & SAVE", sizeSp = 14, color = CourtColors.Ink)
            }
            Box(
                modifier = Modifier.clickable { confirmingCancel = true }.padding(vertical = 2.dp),
            ) {
                LabelText("CANCEL MATCH", color = CourtColors.White.copy(alpha = 0.55f), sizeSp = 12)
            }
        }
    }
}

@Composable
private fun LabeledControl(title: String, content: @Composable () -> Unit) {
    Column(
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(3.dp),
    ) {
        content()
        LabelText(title, color = CourtColors.White45, sizeSp = 9)
    }
}

@Composable
private fun PillButton(label: String, background: Color, text: Color, onClick: () -> Unit) {
    Box(
        modifier = Modifier
            .clip(RoundedCornerShape(20.dp))
            .background(background)
            .clickable { onClick() }
            .padding(horizontal = 14.dp, vertical = 7.dp),
    ) {
        LabelText(label, color = text, sizeSp = 11)
    }
}

/** Two-page swipe indicator, drawn from the shared [Dot]. */
@Composable
private fun PagerDots(count: Int, current: Int, modifier: Modifier = Modifier) {
    Row(modifier = modifier, horizontalArrangement = Arrangement.spacedBy(5.dp)) {
        for (index in 0 until count) {
            Dot(
                sizeDp = 5,
                color = if (index == current) CourtColors.Lime else CourtColors.White.copy(alpha = 0.3f),
            )
        }
    }
}

private enum class ControlIcon { UNDO, PAUSE, PLAY, CLOSE }

/** A round icon button, icon drawn with Canvas — no icon dependency, crisp on the watch. */
@Composable
private fun CircleIconButton(
    icon: ControlIcon,
    onClick: () -> Unit,
    accent: Boolean = false,
    enabled: Boolean = true,
    diameterDp: Int = 36,
) {
    val background = if (accent) CourtColors.Lime else CourtColors.White.copy(alpha = 0.14f)
    val tint = if (accent) CourtColors.Ink else CourtColors.White
    Box(
        modifier = Modifier
            .size(diameterDp.dp)
            .alpha(if (enabled) 1f else 0.35f)
            .clip(CircleShape)
            .background(background)
            .clickable(enabled = enabled) { onClick() },
        contentAlignment = Alignment.Center,
    ) {
        Canvas(Modifier.size((diameterDp * 0.42f).dp)) { drawControlIcon(icon, tint) }
    }
}

private fun DrawScope.drawControlIcon(icon: ControlIcon, color: Color) {
    val w = size.width
    val h = size.height
    val stroke = w * 0.16f
    when (icon) {
        ControlIcon.PAUSE -> {
            val bar = w * 0.26f
            val radius = CornerRadius(bar / 2f)
            drawRoundRect(color, topLeft = Offset(w * 0.16f, 0f), size = Size(bar, h), cornerRadius = radius)
            drawRoundRect(color, topLeft = Offset(w * 0.58f, 0f), size = Size(bar, h), cornerRadius = radius)
        }
        ControlIcon.PLAY -> {
            val path = Path().apply {
                moveTo(w * 0.2f, 0f)
                lineTo(w * 0.92f, h / 2f)
                lineTo(w * 0.2f, h)
                close()
            }
            drawPath(path, color)
        }
        ControlIcon.CLOSE -> {
            drawLine(color, Offset(0f, 0f), Offset(w, h), strokeWidth = stroke, cap = StrokeCap.Round)
            drawLine(color, Offset(w, 0f), Offset(0f, h), strokeWidth = stroke, cap = StrokeCap.Round)
        }
        ControlIcon.UNDO -> {
            // Counterclockwise arc (gap at top-right) + an arrowhead at its start.
            val inset = stroke / 2f
            drawArc(
                color,
                startAngle = -50f,
                sweepAngle = 300f,
                useCenter = false,
                topLeft = Offset(inset, inset),
                size = Size(w - stroke, h - stroke),
                style = Stroke(width = stroke, cap = StrokeCap.Round),
            )
            // Arrowhead at the arc start point (angle -50°), pointing along the sweep.
            val cx = w / 2f
            val cy = h / 2f
            val r = (w - stroke) / 2f
            val a = Math.toRadians(-50.0)
            val tip = Offset(cx + r * cos(a).toFloat(), cy + r * sin(a).toFloat())
            val head = w * 0.28f
            drawLine(color, tip, Offset(tip.x - head, tip.y), strokeWidth = stroke, cap = StrokeCap.Round)
            drawLine(color, tip, Offset(tip.x, tip.y - head), strokeWidth = stroke, cap = StrokeCap.Round)
        }
    }
}

@Composable
private fun TeamRow(modifier: Modifier, short: String, point: String, serving: Boolean) {
    Row(
        modifier = modifier.fillMaxWidth(),
        horizontalArrangement = Arrangement.SpaceBetween,
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Row(verticalAlignment = Alignment.CenterVertically) {
            Dot(sizeDp = 8, visible = serving)
            Spacer(Modifier.width(7.dp))
            DisplayText(short, sizeSp = 20, color = CourtColors.White)
        }
        DisplayText(point, sizeSp = 56, color = CourtColors.White)
    }
}

@Composable
private fun MatchWonScreen(state: MatchState, onEnd: () -> Unit) {
    val won = state.won
    Column(
        modifier = Modifier.fillMaxSize().padding(horizontal = 18.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.Center,
    ) {
        DisplayText("MATCH WON", sizeSp = 24, color = CourtColors.Lime)
        Spacer(Modifier.height(6.dp))
        DisplayText(won?.winnerShort ?: state.teamA.short, sizeSp = 30, color = CourtColors.White)
        Spacer(Modifier.height(6.dp))
        DisplayText(won?.scoreLine ?: state.games, sizeSp = 26, color = CourtColors.White45)
        Spacer(Modifier.height(10.dp))
        LabelText(
            listOf(won?.duration ?: "", "SAVED TO PHONE").filter { it.isNotEmpty() }.joinToString(" · "),
            sizeSp = 10,
        )
        Spacer(Modifier.height(14.dp))
        // DONE lets the user leave MATCH WON — the phone persists the finished
        // match and the next state push lands the watch back on IDLE.
        Box(
            modifier = Modifier
                .clip(RoundedCornerShape(28.dp))
                .background(CourtColors.Lime)
                .clickable { onEnd() }
                .padding(horizontal = 22.dp, vertical = 10.dp),
        ) {
            DisplayText("DONE", sizeSp = 15, color = CourtColors.Ink)
        }
    }
}

File: apps/mobile/targets/watch/Screens.swift
import SwiftUI

// MARK: - Idle

struct IdleView: View {
    let state: MatchState
    let onStartLast: () -> Void

    var body: some View {
        VStack(spacing: 10) {
            Text("NO LIVE MATCH")
                .font(.system(size: 12, weight: .semibold))
                .tracking(1)
                .foregroundStyle(Court.white.opacity(0.5))

            if let last = state.last {
                HStack(spacing: 6) {
                    ResultBadge(won: last.won)
                    Text(last.line)
                        .font(.system(size: 12, weight: .medium))
                        .foregroundStyle(Court.white.opacity(0.55))
                        .lineLimit(1)
                }
            }

            Button(action: onStartLast) {
                VStack(spacing: 1) {
                    Text("START MATCH")
                        .font(.system(size: 15, weight: .heavy))
                    if state.last != nil {
                        Text("LAST LINEUP")
                            .font(.system(size: 9, weight: .semibold))
                            .opacity(0.7)
                    }
                }
                .frame(maxWidth: .infinity)
                .padding(.vertical, 10)
                .foregroundStyle(Court.ink)
                .background(Court.lime, in: Capsule())
            }
            .buttonStyle(.plain)

            Text("or set up on phone")
                .font(.system(size: 10))
                .foregroundStyle(Court.white.opacity(0.4))
        }
        .padding(.horizontal, 16)
        .multilineTextAlignment(.center)
    }
}

// MARK: - Live scoring

/// Apple-Workout-style sideways tabs: swipe between the **controls** (undo,
/// pause, stop-and-save, cancel) and the **scoreboard** (tap a team to score).
/// Separating the two keeps the score face uncluttered and makes an accidental
/// stop impossible while you're rapidly tapping points.
struct LiveView: View {
    let state: MatchState
    let liveBpm: Int
    let onScore: (String) -> Void
    let onUndo: () -> Void
    let onPause: () -> Void
    let onStop: () -> Void
    let onCancel: () -> Void

    // Land on the scoreboard — the surface you touch every rally; controls are
    // one swipe away.
    @State private var tab = Tab.score

    private enum Tab {
        case controls, score
    }

    var body: some View {
        TabView(selection: $tab) {
            ControlsTab(state: state, onUndo: onUndo, onPause: onPause, onStop: onStop, onCancel: onCancel)
                .tag(Tab.controls)
            ScoreTab(state: state, liveBpm: liveBpm, onScore: onScore)
                .tag(Tab.score)
        }
        .tabViewStyle(.page)
    }
}

/// The score face — the only thing on screen is the two tappable team rows.
private struct ScoreTab: View {
    let state: MatchState
    let liveBpm: Int
    let onScore: (String) -> Void

    var body: some View {
        VStack(spacing: 4) {
            header
            TeamScoreRow(
                short: state.teamA.short,
                point: state.pointA,
                serving: state.teamA.serving,
                paused: state.paused,
                onTap: { onScore("A") }
            )
            Rectangle().fill(Court.white.opacity(0.12)).frame(height: 1)
            TeamScoreRow(
                short: state.teamB.short,
                point: state.pointB,
                serving: state.teamB.serving,
                paused: state.paused,
                onTap: { onScore("B") }
            )
        }
        .padding(.horizontal, 14)
        .padding(.top, 2)
    }

    private var header: some View {
        HStack {
            // Live heart rate from the workout session, else the match clock.
            if liveBpm > 0 {
                Text("\(liveBpm)♥")
                    .font(.system(size: 12, weight: .semibold))
                    .foregroundStyle(Court.lime)
            } else {
                Text(state.clock)
                    .font(.system(size: 12, weight: .medium))
                    .foregroundStyle(Court.white.opacity(0.5))
            }
            Spacer(minLength: 4)
            Text([state.setLabel, state.games].filter { !$0.isEmpty }.joined(separator: "  "))
                .font(.system(size: 11, weight: .semibold))
                .foregroundStyle(Court.white.opacity(0.75))
                .lineLimit(1)
            Spacer(minLength: 4)
            if state.paused {
                HStack(spacing: 3) {
                    Circle().fill(Court.white.opacity(0.5)).frame(width: 6, height: 6)
                    Text("PAUSED")
                        .font(.system(size: 10, weight: .bold))
                        .foregroundStyle(Court.white.opacity(0.5))
                }
            } else {
                HStack(spacing: 3) {
                    Circle().fill(Court.lime).frame(width: 6, height: 6)
                    Text("LIVE").font(.system(size: 10, weight: .bold)).foregroundStyle(Court.lime)
                }
            }
        }
    }
}

/// The actions face — undo, pause/resume, the square stop-and-save, and a
/// guarded full cancel.
private struct ControlsTab: View {
    let state: MatchState
    let onUndo: () -> Void
    let onPause: () -> Void
    let onStop: () -> Void
    let onCancel: () -> Void

    @State private var confirmingCancel = false

    var body: some View {
        VStack(spacing: 9) {
            Text(state.paused ? "PAUSED" : "CONTROLS")
                .font(.system(size: 11, weight: .bold))
                .tracking(1.5)
                .foregroundStyle(state.paused ? Court.white.opacity(0.5) : Court.lime)

            HStack(spacing: 18) {
                labeled("UNDO") {
                    CircleIcon(system: "arrow.uturn.backward", tint: Court.white, fill: Court.white.opacity(0.14), action: onUndo)
                        .disabled(state.paused)
                        .opacity(state.paused ? 0.35 : 1)
                }
                labeled(state.paused ? "RESUME" : "PAUSE") {
                    CircleIcon(
                        system: state.paused ? "play.fill" : "pause.fill",
                        tint: Court.ink,
                        fill: Court.lime,
                        diameter: 44,
                        action: onPause
                    )
                }
            }

            // Stop AND save — the square, "court time's up, don't lose the score".
            Button(action: onStop) {
                HStack(spacing: 6) {
                    Image(systemName: "stop.fill").font(.system(size: 13, weight: .bold))
                    Text("STOP & SAVE").font(.system(size: 14, weight: .heavy))
                }
                .frame(maxWidth: .infinity)
                .padding(.vertical, 9)
                .foregroundStyle(Court.ink)
                .background(Court.lime, in: RoundedRectangle(cornerRadius: 12))
            }
            .buttonStyle(.plain)

            // Full cancel discards the match — guarded so a stray tap can't wipe it.
            Button(action: { confirmingCancel = true }) {
                Text("CANCEL MATCH")
                    .font(.system(size: 12, weight: .bold))
                    .foregroundStyle(Court.white.opacity(0.55))
            }
            .buttonStyle(.plain)
            .confirmationDialog("Discard this match?", isPresented: $confirmingCancel) {
                Button("Discard", role: .destructive, action: onCancel)
                Button("Keep playing", role: .cancel) {}
            } message: {
                Text("The score will be lost.")
            }
        }
        .padding(.horizontal, 14)
        .multilineTextAlignment(.center)
    }

    private func labeled<Content: View>(
        _ title: String,
        @ViewBuilder _ content: () -> Content
    ) -> some View {
        VStack(spacing: 3) {
            content()
            Text(title)
                .font(.system(size: 9, weight: .semibold))
                .foregroundStyle(Court.white.opacity(0.5))
        }
    }
}

/// A round icon button — compact and legible on the watch (no wrapping text).
private struct CircleIcon: View {
    let system: String
    let tint: Color
    let fill: Color
    var diameter: CGFloat = 36
    let action: () -> Void

    var body: some View {
        Button(action: action) {
            Image(systemName: system)
                .font(.system(size: diameter * 0.4, weight: .bold))
                .foregroundStyle(tint)
                .frame(width: diameter, height: diameter)
                .background(fill, in: Circle())
        }
        .buttonStyle(.plain)
    }
}

private struct TeamScoreRow: View {
    let short: String
    let point: String
    let serving: Bool
    let paused: Bool
    let onTap: () -> Void

    var body: some View {
        Button(action: onTap) {
            HStack {
                HStack(spacing: 6) {
                    ServingDot(active: serving && !paused)
                    Text(short)
                        .font(.system(size: 17, weight: .bold))
                        .foregroundStyle(Court.white)
                        .lineLimit(1)
                }
                Spacer()
                Text(point)
                    .font(.system(size: 42, weight: .black, design: .rounded).monospacedDigit())
                    .foregroundStyle(Court.white)
            }
            .frame(maxWidth: .infinity, maxHeight: .infinity)
            .contentShape(Rectangle())
        }
        .buttonStyle(.plain)
        .disabled(paused)
        .opacity(paused ? 0.35 : 1)
    }
}

// MARK: - Match won

struct WonView: View {
    let state: MatchState
    let onEnd: () -> Void

    var body: some View {
        VStack(spacing: 5) {
            Text("MATCH WON")
                .font(.system(size: 20, weight: .heavy))
                .foregroundStyle(Court.lime)
            Text(state.won?.winnerShort ?? state.teamA.short)
                .font(.system(size: 26, weight: .black))
                .foregroundStyle(Court.white)
                .lineLimit(1)
            Text(state.won?.scoreLine ?? state.games)
                .font(.system(size: 22, weight: .bold, design: .rounded))
                .foregroundStyle(Court.white.opacity(0.55))
                .lineLimit(1)
            Text([state.won?.duration ?? "", "SAVED TO PHONE"].filter { !$0.isEmpty }.joined(separator: " · "))
                .font(.system(size: 10, weight: .semibold))
                .foregroundStyle(Court.white.opacity(0.5))

            Button(action: onEnd) {
                Text("DONE")
                    .font(.system(size: 14, weight: .heavy))
                    .frame(maxWidth: .infinity)
                    .padding(.vertical, 8)
                    .foregroundStyle(Court.ink)
                    .background(Court.lime, in: Capsule())
            }
            .buttonStyle(.plain)
            .padding(.top, 4)
        }
        .padding(.horizontal, 14)
        .multilineTextAlignment(.center)
    }
}

// MARK: - Small shared pieces

private struct ServingDot: View {
    let active: Bool

    var body: some View {
        Circle()
            .fill(active ? Court.lime : Color.clear)
            .frame(width: 8, height: 8)
    }
}

private struct ResultBadge: View {
    let won: Bool

    var body: some View {
        Text(won ? "W" : "L")
            .font(.system(size: 12, weight: .black))
            .foregroundStyle(won ? Court.ink : Court.white)
            .frame(width: 20, height: 20)
            .background(
                won ? Court.lime : Court.white.opacity(0.15),
                in: RoundedRectangle(cornerRadius: 5)
            )
    }
}
